_posts/2022-12-12-side-projects-2022.md

```diff
@@ -0,0 +1,172 @@
+---
+layout: post
+id: E629FACC-8499-4402-B03B-0F5DB13396BD
+title: The Sparking Joy of Side-Projects, 2022 Edition
+author: Mick F
+excerpt: >-
+  Reviewing 2022, the benefits I gained from running a newsletter as a
+  side-project are tremendous. Let me explain why.
+category: Journaling
+tags:
+  - Swift
+  - iOS
+  - JavaScript
+  - React
+---
+
+🎄 December is here! Like every year, it is time to watch _Home Alone_ again and
+for a review of the progress I made as a developer in the past 12 months.
+Side-projects contributed the most.
+
+## Why side-projects are necessary to me
+
+Yes, a paying job is your bread and butter. But having a side-project is key to
+stay ahead of the younger guns of the tech world, flexing your developer muscles
+in new directions and anticipating what’s next for you at work.
+
+Last year, I started writing [a newsletter about soccer/football][1][^1] for
+people based in France. The content is pretty basic. Still, I learnt so much
+just setting it up, running it, and trying to automate as many steps as possible
+so that each edition would take as little time as possible to write, without
+compromising on its value.
+
+## Things I learned
+
+Here is a non-exhaustive list of the things I trained on last year to run that
+newsletter, in an approximative chronological order:
+
+- [**Swift Argument Parser.**][3] Writing a CLI in Swift is now my goto
+  direction when I need to write scripts. Including at work. It is so much
+  easier to write, document and maintain than Shell scripts.
+- [**Buttondown.**][4] This is an indie tool to bootstrap newsletters. As part
+  of the side-project journey, I always thought it was worth exploring
+  alternatives to big corporations so I skipped Mailchimp, Mailjet and similar
+  options. While I really like it, I think the newsletter is now ready to grow
+  out of it, and finding a better option for my needs is on the roadmap
+  for 2023.
+- [**Linode.**][5] Similarly, I didn’t want to go the AWS route. Since I
+  started, Linode got bought by Akamai so I am not as attached to it as I was.
+  Much clearer than AWS still.
```

Comment:
what does this mean? You still prefer it?

Reply:
I guess so. Yay, I'll try to rephrase to make it clearer.